scheduler/__init__.py

```diff
@@ -1,115 +1,283 @@
+                summarization["equity_name"] = response_json["summary_process"][
+                    "equity_name"
+                ]
+                summarization["financial_type"] = response_json["summary_process"][
+                    "financial_type"
+                ]
+                summarization["remote_blob_url"] = response_json["summary_process"][
+                    "remote_blob_url"
+                ]
+                summarization["summary"] = response_json["summary_process"]["summary"]
         elif success_code == 404:
-            logging.error(f"No new uploaded documents found for: {schedule['companyId']} - {schedule['reportType']}. Last checked time: {start_time}")
+            raise requests.exceptions.RequestException(
+                f"No new documents found for the given equity and report type {schedule['companyId']} - {schedule['reportType']}. Last checked time: {start_time}"
+            )
         else:
-            logging.error(f"Failed to trigger document fetch for: {schedule['companyId']} - {schedule['reportType']}")
-
+            raise requests.exceptions.RequestException(
+                f"Failed to trigger document fetch for {schedule['companyId']} - {schedule['reportType']}"
+            )
     except Exception as e:
-        logging.error(f"Error in trigger_document_fetch: {str(e)}")
-        schedule['summarized_today'] = False
-        return False
-    
+        logging.error(f"Error in [trigger_document_fetch]: {str(e)}")
+        schedule["summarized_today"] = False
+        return summarization
     finally:
         # Always update last run time, regardless of outcome
-        schedule['lastRun'] = start_time
+        schedule["lastRun"] = start_time
         try:
-            cosmos_data_loader.update_last_run(schedule)
+            schedules_db_manager.update_last_run(schedule)
         except Exception as e:
-            raise LastRunUpdateError(schedule.get('id', 'unknown'), e)
+            raise LastRunUpdateError(schedule.get("id", "unknown"), e)
+
+    return summarization
+
+
+def check_subscription_statuses(orgs: List[Dict]) -> List[Dict]:
+    """Check if the subscription is active and it has financial assistant tier"""
+
+    organizations = []
+
+    if len(orgs) == 0:
+        logger.error("No active organizations found")
+        return organizations
+
+    def check_subscription_status(subscription_id: str) -> dict:
+        try:
+            response = requests.get(
+                STRIPE_SUBSCRIPTION_ENDPOINT.replace(
+                    "<subscription_id>", subscription_id
+                ),
+                timeout=TIMEOUT_SECONDS,
+                headers={
+                    "Content-Type": "application/json",
+                    "X-MS-CLIENT-PRINCIPAL-ID": "00000000-0000-0000-0000-000000000000",
+                },
+            )
+            response_json = response.json()
+
+            if "subscriptionData" not in response_json:
```

Comment:
@PedroLabrador If response_json is None (due to a failed request), accessing response_json["subscriptionData"] will cause an uncaught TypeError.

Validate response_json before accessing it:
```python
if not response_json or not isinstance(response_json, dict):
    raise requests.exceptions.RequestException("Invalid response received")
 ```

Reply:
response_json would never be None (due to a failed request), if the request fails, we have a try catch (requests.exceptions.RequestException) for that

Reply:
Because if requests fails raises an exception before I can parse the result